### PR TITLE
ETH JSON RPC and Brave native wallet experimental flag support

### PR DESCRIPTION
**Pull request by bbondy**
Resolves https://github.com/brave/brave-browser/issues/14250
Resolves https://github.com/brave/brave-browser/issues/14251

- This introduces a feature flag that we will use to gate native wallet features to be added later.
- It also starts to implement the most basic code to do a JSON RCP Ethereum call on Infura mainnet

It is easiest to review this 1 commit at a time.

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)
- [ ] Requested a security/privacy review as needed

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/eth_json_rpc_controller.cc [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    SimpleURLLoaderList::iterator iter,
+    URLRequestCallback callback,
+    const std::unique_ptr<std::string> response_body) {
+  auto* loader = iter->get();
+  auto response_code = -1;
+  std::map<std::string, std::string> headers;
+  if (loader->ResponseInfo() && loader->ResponseInfo()->headers) {
+    response_code = loader->ResponseInfo()->headers->response_code();
+    auto headers_list = loader->ResponseInfo()->headers;
+    if (headers_list) {
```
Comment (SergeyZhukovsky) [inline components/brave_wallet/eth_json_rpc_controller.cc:95]: perhaps we don't need that if as we checked on null in the above if

File: components/brave_wallet/eth_json_rpc_controller.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      base::OnceCallback<void(const int,
+                              const std::string&,
+                              const std::map<std::string, std::string>&)>;
+  void Request(const std::string& json_payload,
+               URLRequestCallback callback,
+               bool auto_retry_on_network_change);
+
+ private:
+  using SimpleURLLoaderList =
+      std::list<std::unique_ptr<network::SimpleURLLoader>>;
```
Comment (SergeyZhukovsky) [inline components/brave_wallet/eth_json_rpc_controller.h:43]: missed memory header

**Review by mkarolin — APPROVED** [review]
`chromium_src` changes LGTM

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Pull request merged**